cloudinit/net/network_state.py

```diff
@@ -303,6 +304,10 @@ def parse_config_v1(self, skip_broken=True):
                 ) from e
             try:
                 handler(self, command)
+                if command_type == 'physical':
```

Comment:
Unfortunately, this isn't going to work. The v1 config is unorderered so a nameserver can't correspond directly to a specific interface. This also fails to parse if we have a config like this:
```
network:
  version: 1
  config:
    - type: nameserver
      address:
        - 192.168.23.2
        - 8.8.8.8
      search:
        - exemplary
    - type: physical
      name: eth0
      mac_address: '00:11:22:33:44:55'
      subnets:
         - type: static
           address: 192.168.23.14/27
           gateway: 192.168.23.1
```

I wouldn't be opposed to expanding the v1 config to add an 'interface' key to the 'nameserver' type, but I think we'd be better served doing that in a separate PR.

Reply:
I understand, I was being optimistic about the order of configurations. What can we do about in this PR?

Reply:
I see 3 options:
1. Create and wait for the other PR to land and rebase it into this one
2. Apply the globally defined dns entries to every interface (this is what netplan does)
3. Disallow the networkd renderer from using v1 config until we can make the change

Reply:
I have decided to use option 2 for the time being. I will start working on adding new option net config v1.